cdap-docs/tools/docs-cli-commands.py

```diff
@@ -0,0 +1,171 @@
+
+def skip_this_section(line):
+    text = line.strip(' *').lower()
+    return text in SKIP_SECTIONS
+    
+def create_literal_line(line):
+    right_line = line.lstrip()
+    spaces = len(line) - len(right_line)
+    return LITERAL_LINE + ' ' * spaces + LITERAL + right_line + LITERAL
+
+def create_parsed_line(line):
+    """ Parses a line and changes single-quotes to a combination of literals and single-quotes.
+        Handles closing literals with a backslash correctly for alpha-numerics.
+        Any text that is enclosed with single-quotes will be converted to a literal with single-quotes.
+        'text' will become ``'text'``
+    """
+    i = 0
+    in_literal = False
+    finished_literal = False
+    literal = '``'
+    quote = '\''
+    backslash = '\\'
+    opening_literal_quote = literal + quote
+    closing_literal_quote = quote + literal
+    new_line = ''
+    for i in range(len(line)):
+        c = line[i]
+        if c == quote:
+            if not in_literal:
+                new_line += opening_literal_quote
+            else:
+                new_line += closing_literal_quote
+                finished_literal = True
+            in_literal = not in_literal
+        else:
+            if finished_literal:
+                if c.isalnum():
+                    new_line += backslash
+                finished_literal = False
+            new_line += c
+    return new_line
+
+#
+# Create the table
+#
+
+def create_table(input_file, output_file):
+    """ Creates the table by reading in a file, building the table, outputing to a new file
+    """
+    lines = []
+    return_code = 0
+    if os.path.isfile(input_file):
+        print "Reading in %s" % input_file
+        f = open(input_file,'r')
+        in_spaces = False
+        in_literal = False
+        skip_section = False
+        for line in f:
+            line = line.rstrip()
+            if line_starts_with(line, SECTION_LINE) or line_starts_with(line, COMMAND_LINE):
+                if line_starts_with(line, SECTION_LINE):
+                    skip_section = skip_this_section(line)
+                if skip_section:
+                    continue             
+                if in_spaces: # Insert a blank line
+                    lines.append('')
+                    in_spaces = False
+                lines.append(create_parsed_line(line))
+            elif skip_section:
+                continue
+            elif line_starts_with(line, SPACES): # Handle spaces: assume a literal
+                in_spaces = True
+                lines.append(create_literal_line(line))
+            else:
+                if in_spaces: # Insert a blank line
+                    lines.append('')
+                    in_spaces = False
+                lines.append(SPACES + create_parsed_line(line))
+    else:
+        print "Did not find %s" % input_file
+        print "Wrote 'missing file' to table"
+        lines.append(MISSING_FILE_TEMPLATE % input_file)
```

Comment:
We write to the output file even if there is no input so that the Sphinx build will still succeed and we will get the rest of the output. I didn't want a fatal build error.